packages/components/src/__overlays__/Tooltip/v3/Tooltip.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { forwardRef, useContext, useLayoutEffect, useState } from "react"
+import { VisuallyHidden } from "react-aria"
+import {
+  Tooltip as RACTooltip,
+  TooltipContext,
+  TooltipProps as RACTooltipProps,
+  TooltipTriggerStateContext,
+  useContextProps,
+} from "react-aria-components"
+import { useReversedColors } from "~components/__utilities__/v3"
+import { mergeClassNames } from "~utils/mergeClassNames"
+import { OverlayArrow } from "./OverlayArrow"
+import styles from "./Tooltip.module.scss"
+
+export { TooltipContext }
+
+export type TooltipProps = Omit<RACTooltipProps, "offset"> & {
+  /**
+   * The additional offset applied along the main axis between the element and its
+   * anchor element.
+   * @default 8
+   */
+  offset?: number
+  /**
+   * The placement of the tooltip relative to the trigger.
+   * + we're intentionally limiting the placement options to rtl friendly ones (start, end, top, bottom)
+   */
+  placement?:
+    | "bottom"
+    | "bottom start"
+    | "bottom end"
+    | "top"
+    | "top start"
+    | "top end"
+    | "start"
+    | "start top"
+    | "start bottom"
+    | "end"
+    | "end top"
+    | "end bottom"
+}
+
+/**
+ * A tooltip displays a description of an element on hover or focus.
+ */
+export const Tooltip = forwardRef<HTMLDivElement, TooltipProps>(
+  ({ children, className, ...props }, ref): JSX.Element => {
```

Comment:
Realised this is `className` instead of `classNameOverride`, I'm good with the rename to `className`, but I think we should keep it consistently `classNameOverride` for now and go through the deprecation process to change all the components at once.

Reply:
Yes this is intentional change. With v3 (and whole new gen of components) we aim to cleanup a lot of custom interfaces and patterns that are across the components and fully clean them up. No more moods, no more `classNameOverride` and other. 
The consistency goal is that this will be pattern for all new/v3+ components. There's a bit of question what we do when we start v4 and backfill v3 from v2 if we rename it everywhere or just move, but that's bit future problem.